repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,158 @@
+            api.current_logger().error("Cannot perform the upgrade because there is\
+             no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop after upgrade."),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([
+                    reporting.Tags.UPGRADE_PROCESS
+                ]),
+                reporting.Flags([
+                    reporting.Flags.INHIBITOR
+                ]),
+                reporting.Remediation(
+                    hint="Install GNOME desktop to be able to upgrade.")
+                ])
+            return
+
+        # Assume GNOME is installed in this state
+
+        user = os.environ.get("USER")
+        default_xsession = get_xsession("/var/lib/AccountsService/users/{0}".format(user))
+        if not default_xsession:
+            api.current_logger().warn("Unable to detect default session.")
+        else:
+            if "plasma" in default_xsession:  # using in because there can be some white spaces
+                api.current_logger().info("KDE used as default session.")
+                api.current_logger().info("Upgrade can be performed, but KDE desktop will\
+                 be removed in favor of GNOME")
+                reporting.create_report([
+                    reporting.Title("Upgrade can be performed, but KDE will be uninstalled."),
+                    reporting.Summary("KDE has to be uninstalled in favor of GNOME to perform the upgrade."),
+                    reporting.Severity(reporting.Severity.MEDIUM),
+                    reporting.Tags([
+                        reporting.Tags.UPGRADE_PROCESS
+                    ]),
+                    reporting.Remediation(
+                        hint="KDE has to be removed, no other solution is possible.")
+                    ])
+            else:
+                api.current_logger().info("GNOME used as default session. Continuing with the upgrade.")
+        api.current_logger().info("----------------------------------")
+
+    # At this state we can assume that KDE desktop as such is not installed or used and we just need to
+    # detect whether any KDE/Qt app is actively used to inform user that the application will be removed
+    # during the upgrade process
+
+    base_kde_apps = ("kde-baseapps",
+                     "okular",
+                     "ark",
+                     "kdepim",
+                     "konsole",
+                     "gwenview",
+                     "kdenetwork",
+                     "kate", "kwrite")
+    api.current_logger().info("  Detecting installed KDE apps  ")
+    api.current_logger().info("================================")
+
+    for app in base_kde_apps:
```

Comment:
Nit: 
`for app in [app for app in base_kde_apps if is_installed(app) and is_in_use(app)]`

instead of cycles at L135-L136